core/src/identity.rs

```diff
@@ -1,28 +1,27 @@
 use serde::{Deserialize, Serialize};
 
-#[derive(Clone, Debug, Copy)]
-pub struct Identity<T> {
-    resolver: T,
-}
+pub struct Identity {}
```

Comment:
We should delete this struct altogether as it's necessary. The user can just call the `recover` function directly

Reply:
@sKudryashov The change to make recover stateless looks good but we should still delete the `Identity` struct as it's not needed at least from what I see